### ISSUE TITLE
Object segmentation 

### ISSUE THREAD

**Issue by 704051419** [j96w/DenseFusion#120]
I'm using the Linemod dataset.
the workflow includes the "object segmentation "part, but when training, the color embeddings are from the mask of the training set. so can I say, you use the ground truth "mask" as the  results of object segmentation, and I notice the mask includes the  occlusion part. does it affect the  performance ? 
In article it mentions that the segmentation part use the same architecture with PoseCNN , so I get a little confused.
And when changing the segmentation part ,will it the performance largely  boost ? cause initiatively, the pointcloud and the geometry embeddings are more important.

**Comment by 704051419** [comment]
![20180820173343974](https://user-images.githubusercontent.com/34223282/68528599-ddef1180-032f-11ea-85aa-965660e62824.png)
another question is about PointNet, in the `network .py`, I can't correspond the architecture to the original architecture of pointNet, can you figure it out ?
Thanks !

**Comment by j96w** [comment]
Hi,

(1) Segmentation: The ground truth segmentation is only used in the training process. In evaluation, for comparison with PoseCNN, we test our model based on the same outputted segmentation results from the trained SegNet model of PoseCNN.

(2) PointNet: Please refer to [Issue#34](https://github.com/j96w/DenseFusion/issues/34).

**Comment by 704051419** [comment]
@j96w  i mean, when the mask corresponding to the occlusion part is included, the RGB information and the depth from the images are actuality not the real information about the target object, because the real RGB and depth are occlusied. is that okay?

**Comment by 704051419** [comment]
@j96w another question is: the geometry embedding process includes no pooling, so how to ensure that the order of the pointcloud doesn't affect the performance?
although when calculate the global feature, there is an average pooling, but then when `return torch.cat([pointfeat_1, pointfeat_2, ap_x], 1) #128 + 256 + 1024`, `pointfeat1`and `pointfeat2`include the geometry feature, is that okay ?

**Comment by 704051419** [comment]
@j96w could you tell me how can I get the segment result of the training data of Linemod. Or where is the trained SegNet for it ?
Thanks !

**Comment by twinsnan0** [comment]
> @j96w could you tell me how can I get the segment result of the training data of Linemod. Or where is the trained SegNet for it ?
> Thanks !

@704051419 Have you figure it out?
